ams/server/src/main/java/com/netease/arctic/server/optimizing/scan/IcebergTableFileScanHelper.java

```diff
@@ -31,12 +31,14 @@
 import org.apache.iceberg.Table;
 import org.apache.iceberg.io.CloseableIterable;
 import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.io.IOException;
 import java.io.UncheckedIOException;
 import java.util.List;
+import java.util.Map;
 import java.util.stream.Collectors;
 
 public class IcebergTableFileScanHelper implements TableFileScanHelper {
```

Comment:
I'm not sure if KeyedTableFileScanHelper and UnkeyedTableFileScanHelper will have this situation, because I don't have a corresponding environment locally.

Reply:
I think this problem will be much better in mixed-format, because in mixed-format, delete files are organized by node hierarchy and will not intersect with many data files. And when planning, tasks are also organized by nodes, not files like in an iceberg table.